man/man3/fi_mr.3

```diff
@@ -503,18 +503,23 @@ fi_mr_attr (defined below).
 .PP
 Fi_close is used to release all resources associated with a registering
 a memory region.
-Once unregistered, further access to the registered memory is not
+Once deregistered, further access to the registered memory is not
 guaranteed.
 Active or queued operations that reference a memory region being closed
-may fail or result in accesses to invalid memory.
+may attempt to access an invalid memory region and fail.
+After an MR is closed, any new operations targeting the closed MR will
+also fail.
```

Comment:
@shefty is this really true / guaranteed ? default behavior with verbs / cxi is that operations targeting MRs that remain exposed in the MR cache may just succeed.

Reply:
This is the behavior that customers need.  I'm surprised that you're putting explicitly created MRs in the registration cache, for exactly the security reason.  Implicitly created MRs makes complete sense to put in the cache, but I don't understand how you can have a transparent cache and allow customers their security model.

Reply:
Actually, for the OFI NCCL Plugin, we tried pretty hard not to let Libfabric have things in a cache.  And more than one group is messing with fault tolerance under NCCL, which really assumes killing an MR kills communication.  Hence the path that lead us to these questions.

Reply:
I agree this is also what I would want  as a user so I can rely on those semantics but this has not been the default so if this is the new default, providers should update their code accordingly at the detriment of performance or maybe there should be a note in that paragraph that explains why this may not be guaranteed for performance reasons.

Reply:
I like the current wording that you are proposing. The only concern I was raising was that some providers are currently not conforming to this behavior and that can be misleading for application developers. So my question was more, if we are adding that line, are we expecting providers to be updated to reflect that ? If not then there should be an additional note that warns developers.